packages/mocks/webpack.config.ts

```diff
@@ -0,0 +1,117 @@
+/* eslint-disable @cspell/spellchecker */
+/* eslint-env node */
+
+import * as path from 'path';
+
+import CopyPlugin from 'copy-webpack-plugin';
+import TerserPlugin from 'terser-webpack-plugin';
+import { TsconfigPathsPlugin } from 'tsconfig-paths-webpack-plugin';
+import { type Configuration as WebpackConfiguration, EnvironmentPlugin } from 'webpack';
+import { type Configuration as WebpackDevServerConfiguration } from 'webpack-dev-server';
+
+import { DynamicConsoleRemotePlugin } from '@kubev2v/webpack';
+
+import pluginMetadata from './plugin-metadata';
+
+const pathTo = (relativePath: string) => path.resolve(__dirname, relativePath);
+
+const production = false; // process.env.NODE_ENV === 'production';
+const configFile = pathTo('tsconfig.json');
+
+const config: WebpackConfiguration & {
+  devServer: WebpackDevServerConfiguration;
+} = {
+  mode: production ? 'production' : 'development',
+  context: pathTo('src'),
+  entry: {}, // entry generated by `DynamicConsoleRemotePlugin`
+  output: {
+    path: pathTo('dist'),
+    publicPath: `/api/plugins/${pluginMetadata.name}/`,
+    chunkFilename: production ? 'chunks/[name].[chunkhash:8].min.js' : 'chunks/[name].js',
+    assetModuleFilename: production ? 'assets/[name].[contenthash:8][ext]' : 'assets/[name][ext]',
+    filename: production ? '[name]-bundle-[hash:8].min.js' : '[name]-bundle.js',
+  },
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js', '.jsx'],
+    plugins: [new TsconfigPathsPlugin({ configFile })],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.(jsx?|tsx?)$/,
+        exclude: /node_modules/,
+        use: [
+          {
+            loader: 'ts-loader',
+            options: {
+              configFile,
+            },
+          },
+        ],
+      },
+      {
+        test: /\.s?(css)$/,
+        use: ['style-loader', 'css-loader'],
+      },
+      {
+        test: /\.(png|jpg|jpeg|gif|svg|woff2?|ttf|eot|otf)(\?.*$|$)/,
+        type: 'asset/resource',
+        generator: {
+          filename: 'assets/[name].[ext]',
+        },
+      },
+      {
+        test: /\.m?js/,
+        resolve: {
+          fullySpecified: false,
+        },
+      },
+    ],
+  },
+  devServer: {
+    static: './dist',
+    host: 'localhost',
+    port: 9002,
+    devMiddleware: {
+      writeToDisk: true,
+    },
+
+    // Allow bridge running in a container to connect to the plugin dev server.
+    allowedHosts: 'all',
+    headers: {
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, PATCH, OPTIONS',
+      'Access-Control-Allow-Headers': 'X-Requested-With, Content-Type, Authorization',
+      'Service-Worker-Allowed': '/', // only needed on service worker scripts to allow the MSW to scope to app root
```

Comment:
It's very nice that the Console forwards the header! The only problem is that this is undocumented feature.

Reply:
Indeed, that would be a nice doc on bridge!

See use of `proxy.FilterHeaders(resp)` in the bridge plugin handler:
https://github.com/openshift/console/blob/master/pkg/plugins/handlers.go#L150-L193

And the actual filter code:
https://github.com/openshift/console/blob/d1fd2d8f2247a15096496ee89ca5eb228b03ceed/pkg/proxy/proxy.go#L35-L53

Reply:
(moved to forklift-console-plugin's webpack dev server)